kafka-rest/src/main/java/io/confluent/kafkarest/Utils.java

```diff
@@ -67,4 +67,23 @@ public static int errorCodeFromProducerException(Throwable e) {
     }
     return Response.Status.OK;
   }
+
+  /**
+   * Tries up to {@code maxAttempts} times to get the value from the {@code supplier}, retrying if
```

Comment:
i haven't investigated use of this fully yet, but it smells a bit bad to me.

Reply:
Can you elaborate please? :)

Reply:
the purpose is to repeatedly run code (without any backoff) until an error stops occurring. that seems a very dubious idea to me, and unlikely to justifiable.